### PR TITLE
Split api test case into two contexts json and xml

### PR DESCRIPTION
**Pull request by Arminek**

### CODE REVIEW

File: src/ApiClientType.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Lakion\ApiTestCase;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class ApiClientType
```
Comment (pjedrzejewski) [inline src/ApiClientType.php]: I'd say ``MediaTypes``.

https://en.wikipedia.org/wiki/Media_type

**Comment by pjedrzejewski** [comment]
Nice! :+1: 

Throwing random discussion:

XmlApiTestCase vs. ApiXmlTestCase
JsonApiTestCase vs. ApiJsonTestCase

To me, first variant sounds more natural :) Thoughts?

**Comment by Arminek** [comment]
IMO first variant.

File: src/MediaTypes.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Lakion\ApiTestCase;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class MediaTypes
```
Comment (pamil) [inline src/MediaTypes.php]: I'm not sure about this, why don't we store MIME types in `*ApiTestCase`?

**Comment by pamil** [comment]
Got some ideas:
 - let's rename `assert(?:Json|Xml)` methods to `assert` (regexp overflow :D I mean `assertJsonResponse` to `assertResponse` etc.)
 - don't include `.json` or `.xml` extensions in `$filename`
 - specific `*ApiTestCase` will take care of appending appropriate extension, resolving MIME types etc.

This way we only need to add XML responses and change our parent ApiTestCase to XML one and we can be testing XML api instead of JSON one.

Moreover, we can extract asserting logic to interface like `AssertionEngineInterface` and allow to register many assertion engines in one ApiTestCase, this will let us test XML and JSON api at the same time :)

Just sayin'! :tada:

File: src/MediaTypes.php (hunk 2 above)
Comment (Arminek) [inline src/MediaTypes.php]: I think about some kind of extension guesser, that's why i create separate class.

**Comment by Arminek** [comment]
I am not sure that we need to test different response types in one test scope.

**Comment by pjedrzejewski** [comment]
@Arminek I think @pamil's idea is to avoid coupling a test case to specific media type. Only indicator if we are testing Xml or Json would be the base test case class and that's it. :+1: from me.

I'd leave the idea of AssertionEngineInterface for v2 or something. :)

**Comment by lchrusciel** [comment]
:+1: for AssertionEngineInterface, but in v2. XmlApiTestCase vs. ApiXmlTestCase? I would say XmlTestCase is enough. But from this two I would choose XmlApiTestCase.

**Comment by pjedrzejewski** [comment]
:+1:

**Pull request merged**